ocaml/typing/typetexp.ml

```diff
@@ -1020,10 +1026,10 @@ and transl_type_alias env policy mode alias_loc styp name_opt layout_annot_opt =
       with Not_found ->
         if !Clflags.principal then begin_def ();
         let layout =
-          Layout.(of_annotation_option_default
-            ~default:(any ~why:Dummy_layout)
-            ~context:(Type_variable alias)
-            layout_annot_opt)
+          Option.fold ~none:(Layout.any ~why:Dummy_layout)
```

Comment:
I think this would be easier to read with a `match layout_annot_opt with ...` construction instead of `Option.fold`.